blessed/keyboard.py

```diff
@@ -48,40 +89,606 @@ class attributes for comparison, such as ``term.KEY_LEFT``.
+            else:
+                # Remaining tokens could be a key name
+                tokens_key_names = tokens[i:]
+                break
+
+        # If we have any non-modifier tokens,
+        if tokens_key_names:
+            # check if they form a valid application key,
+            key_name = '_'.join(tokens_key_names)
+            keycodes = get_keyboard_codes()
+            expected_key_constant = f'KEY_{key_name.upper()}'
+            if expected_key_constant in keycodes.values():
+                # and return as predicate function
+                return self._build_appkeys_predicate(tokens_modifiers, key_name)
+
+        # No valid key name was found by 'tokens_key_names', this could just as
+        # easily be asking for an attribute that doesn't exist, or a spelling
+        # error of application key or modifier, report as 'invalid' token
+        invalid_tokens = [token for token in tokens
+                          if token not in KittyModifierBits.names_modifiers_only]
+        if invalid_tokens:
+            raise AttributeError(
+                f"'{self.__class__.__name__}' object has no attribute '{attr}' "
+                f"(invalid modifier or application key tokens: {invalid_tokens})")
+
+        # Return modifier predicate for alphanumeric keys
+        return self._build_alphanum_predicate(tokens_modifiers)
+
+    def _get_plain_char_value(self) -> Optional[str]:
+        """
+        Get value for plain printable characters.
+
+        Returns the character as-is if it's a single printable character.
+        """
+        if len(self) == 1 and not self[0] == '\x1b' and self[0].isprintable():
+            return str(self)
+        return None
+
+    def _get_alt_sequence_value(self) -> Optional[str]:
+        """
+        Get value for Alt+printable sequences (ESC + char).
+
+        Returns the printable character from Alt sequences.
+        """
+        if (len(self) == 2 and self[0] == '\x1b' and
+                self._alt and not self._ctrl and self[1].isprintable()):
+            return self[1]  # Return as-is (preserves case and supports Unicode)
+        return None
+
+    def _get_alt_control_sequence_value(self) -> Optional[str]:
+        """
+        Get value for Alt-only control sequences (ESC + control char, Alt-only).
+
+        Returns empty string for special application keys with Alt modifier.
+        """
+        if not (len(self) == 2 and self[0] == '\x1b' and
+                self._alt and not self._ctrl):
+            return None
+
+        char_code = ord(self[1])
+
+        # Special application keys with Alt modifier should return empty string
+        # These are: Escape (0x1b), Backspace/DEL (0x7f), Enter (0x0d), Tab (0x09)
+        # They are application keys, not text keys, so they have no text value
+        if char_code in {0x1b, 0x7f, 0x0d, 0x09}:
+            return ''
+
+        return None
+
+    def _get_ctrl_alt_sequence_value(self) -> Optional[str]:
```

Comment:
this method, _get_ctrl_alt_sequence_value and previous, _get_alt_control_sequence_value are combined in #309 as _get_escape_sequence_value slightly reducing LOC